### PR TITLE
Refactor CPUID code, take 2

### LINKED ISSUE #11281: Build OpenSSL on ubuntu, compile failed in x86_32(providers/legacy.so)
Get the latest OpenSSL code of master branch, "./config; make; make install_sw;". compile it on ubuntu. find that the builds on x86-64,aarch32,aarch64 are success but fails on x86-32. The error log as below:
```
rm -f test/dsa_no_digest_size_test
${LDCMD:-gcc} -pthread -m32 -Wa,--noexecstack -Wall -O3 -fomit-frame-pointer -Werror   \
	-o test/dsa_no_digest_size_test \
	test/dsa_no_digest_size_test-bin-dsa_no_digest_size_test.o \
	test/libtestutil.a libcrypto.a -ldl -pthread 
rm -f test/dsatest
${LDCMD:-gcc} -pthread -m32 -Wa,--noexecstack -Wall -O3 -fomit-frame-pointer -Werror   \
	-o test/dsatest \
	test/dsatest-bin-dsatest.o \
	test/libtestutil.a libcrypto.a -ldl -pthread 
collect2: error: ld returned 1 exit status
Makefile:21332: recipe for target 'providers/legacy.so' failed
make[1]: *** [providers/legacy.so] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/ent-user/ci-scripts/openssl-src'
Makefile:2899: recipe for target 'build_sw' failed
make: *** [build_sw] Error 2

```

### PR DESCRIPTION
**Pull request by levitte**
We were using CPUID coded in several modules, but it was unclear how
it actually got there, and could fail randomly.

To remedy that, this change separates the CPUID C code from the rest
of cryptlib.c, and ensures the right modules get both that and the
assembler sources explicitly.

Fixes #11281

### CODE REVIEW

**Comment by levitte** [comment]
Judging from the current build status for https://github.com/levitte/openssl/tree/fix-11281%2B%2Bworkflow-reimplement-no-shared (which now also includes @slontis' #14738), this PR is good to go now.

**Comment by levitte** [comment]
The external-tests failure currently visible doesn't look relevant...

File: crypto/cpuid.c [hunk 1]
```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 1998-2020 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (slontis) [inline crypto/cpuid.c:2]: Not sure why you went backwards with the date here :)

Comment (levitte) [reply crypto/cpuid.c:2]: I didn't...  I originally did this stuff last year, and didn't think to look now.

Comment (slontis) [reply crypto/cpuid.c:2]: I would argue that you added this file just yesterday (1998-2021)

Comment (levitte) [reply crypto/cpuid.c:2]: Fixed

Comment (levitte) [reply crypto/cpuid.c]: > I would argue that you added this file just yesterday (1998-2021)

Then you would be wrong.

> commit c1c52b522572327fca4972bacef37e032e0a73ed
> Author: Richard Levitte <levitte@openssl.org>
> Date:   Wed Mar 11 17:38:46 2020 +0100

**Review by t8m — APPROVED** [review]
Good work!

**Comment by levitte** [comment]
ef83daf4dadf9380a3b94618fb7aee75fcd9a6b1 Refactor CPUID code
5ad3e6c56eb1c295a7de92de5bb2f54614d5c277 Include BN assembler alongside CPUID code

**Pull request closed without merging**